File: apps/onework-manage-api/app/model/database/column.ts
/*
 * @Author: 钟凯
 * @Date: 2021-03-01 14:58:53
 * @LastEditTime: 2021-03-31 15:37:49
 * @LastEditors: 钟凯
 * @Description:
 * @FilePath: \egg_ts\app\model\database\column.ts
 * @可以输入预定的版权声明、个性签名、空行等
 */
import { DataTypes } from 'sequelize';
import { Application, Ow, Sequelize } from 'egg';
import { NameCodeModel } from '../../core/index';

export default (app:Application) => {

  const Column = app.model.define<Sequelize.Database.Column, Ow.Database.Column>('column', {
    ...NameCodeModel,
    cnUid: { type: DataTypes.STRING, allowNull: false, comment: '连接uid' },
    dbUid: { type: DataTypes.STRING, allowNull: false, comment: '数据库uid' },
    tbUid: { type: DataTypes.STRING, allowNull: false, comment: '数据库表uid' },
    dbName: { type: DataTypes.STRING, allowNull: false, comment: '数据库uid' },
    type: { type: DataTypes.TEXT, allowNull: false, comment: '类型' },
    isNull: { type: DataTypes.BOOLEAN, allowNull: false, comment: '是否为空' },
    isUnique: { type: DataTypes.BOOLEAN, allowNull: false, comment: '是否主键' },
    length: { type: DataTypes.BIGINT, allowNull: true, comment: '长度' },
    precision: { type: DataTypes.INTEGER, allowNull: true, comment: '精度' },
    oredr: { type: DataTypes.INTEGER, allowNull: true, comment: '序号' },
    defaultValue: { type: DataTypes.TEXT, allowNull: true, comment: '默认值' },
    description: { type: DataTypes.TEXT, allowNull: true, comment: '描述' },

  }, {
    tableName: 'ow_database_columns',
    paranoid: true,
  });
  return Column;
};

File: apps/onework-tools/onework-tools-domain/src/main/java/com/onework/tools/domain/database/dao/Column.java
package com.onework.tools.domain.database.dao;

import cn.hutool.core.bean.BeanUtil;
import com.onework.tools.domain.database.schema.entity.DataColumn;
import lombok.Data;

/**
 * @author : zhongkai1010@163.com
 * @version V1.0
 * @Project: onework-tools
 * @Package com.onework.tools.domain.database.dao
 * @Description: 描述
 * @date Date : 2022年03月29日 16:31
 */

@Data
public class Column {

    private String uid;
    private String name;

    private String cnUid;
    private String dbUid;
    private String dbName;
    private String tbUid;
    private String tbName;

    private String type;
    private Long length;
    private Integer precision;
    private Boolean isNull;
    private Boolean isUnique;
    private Integer order;
    private String defaultValue;
    private String description;

    public static Column getColumn(DataColumn dataColumn, Table table) {

        Column column = BeanUtil.copyProperties(dataColumn, Column.class);
        column.setCnUid(table.getCnUid());
        column.setDbUid(table.getDbUid());
        column.setDbName(table.getDbName());
        column.setTbUid(table.getUid());
        column.setTbName(table.getName());
        return column;
    }
}
